undefined reference to boost::filesystem::detail::copy_file

As reported by @fluffypony and @majestrate:

    ./libi2pd-core.a(util.cpp.o): In function `boost::filesystem::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option)':
    /usr/include/boost/filesystem/operations.hpp:381: undefined reference to boost::filesystem::detail::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option, boost::system::error_code*)'

Problem happens with boost 1.55 and gcc 4.8.4.
Does not happen with gcc 5.2.1 or MSVC and boost 1.58.

Possible workaround?

    #define BOOST_NO_CXX11_SCOPED_ENUMS

--- a/core/AddressBook.cpp
+++ b/core/AddressBook.cpp
@@ -466,7 +466,7 @@ namespace client
         bool success = false;   
         i2p::util::http::url u (m_Link);
         i2p::data::IdentHash ident;
-        if (m_Book.GetIdentHash (u.host_, ident))
+        if (m_Book.GetIdentHash (u.host_, ident) && m_Book.getSharedLocalDestination())
         {
             std::condition_variable newDataReceived;
             std::mutex newDataReceivedMutex;

--- a/core/util/util.cpp
+++ b/core/util/util.cpp
@@ -1,3 +1,4 @@
+#include "util.h"
 #include <cstdlib>
 #include <string>
 #include <algorithm>
@@ -12,7 +13,6 @@
 #include <boost/program_options/detail/config_file.hpp>
 #include <boost/program_options/parsers.hpp>
 #include <boost/algorithm/string.hpp>
-#include "util.h"
 #include "Log.h"
 
 #if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__APPLE__) || defined(__OpenBSD__)

--- a/core/util/util.h
+++ b/core/util/util.h
@@ -1,6 +1,8 @@
 #ifndef UTIL_H
 #define UTIL_H
 
+#define BOOST_NO_CXX11_SCOPED_ENUMS // Workaround for issue #272
+
 #include <map>
 #include <string>
 #include <iostream>

